snapshot/snapshot.go

```
@@ -76,19 +77,24 @@ func NewSnapshotter(ctx context.Context, cfg *config.Config) (snapshots.Snapshot
 		return nil, errors.Wrap(err, "parse recover policy")
 	}
 
-	pm, err := manager.NewManager(manager.Opt{
+	manager, err := manager.NewManager(manager.Opt{
 		NydusdBinaryPath: cfg.NydusdBinaryPath,
 		Database:         db,
 		DaemonMode:       cfg.DaemonMode,
 		CacheDir:         cfg.CacheDir,
 		RecoverPolicy:    rp,
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to new process manager")
+		return nil, errors.Wrap(err, "create daemons manager")
+	}
+
+	if cfg.APISocket != "" {
+		systemController, _ := system.NewSystemController(manager, cfg.APISocket)
```

Why ignore error here?

---

fixed